types/sinon/ts3.1/index.d.ts

```diff
@@ -404,7 +404,7 @@ declare namespace Sinon {
          * The Promise library can be overwritten using the usingPromise method.
          * Since sinon@2.0.0
          */
-        resolves(value?: TReturnValue extends PromiseLike<infer TResolveValue> ? TResolveValue : never): SinonStub<TArgs, TReturnValue>;
+        resolves(value?: TReturnValue extends PromiseLike<infer TResolveValue> ? TResolveValue : any): SinonStub<TArgs, TReturnValue>;
```

Comment:
but this now means that if you do pass in the wrong type, the whole thing falls back to any and you lose all strong typing.

i.e. if your stub returns `Promise<number>` and you do:

```typescript
myStub.resolves('hello');
```

Now the inference will fail, and this falls back to `'hello'` which means this is broken.

What happens if you resolve to `unknown` instead, then you could at least cast bad values:

```typescript
myStub.resolves('hello' as unknown);
```

Reply:
no, it will not.

`TReturnValue` is from the type parameters of the stub. You can't pass a string in, `TReturnValue` is `Promise<number>` which extends `PromiseLike<number>` which means `resolves(value?: number)` here.

Reply:
there you go i've even added a test now to show you this is the case. does it make more sense now?